### ISSUE TITLE
struct members should be snake_case

### ISSUE THREAD

**Issue by clux** [kube-rs/kopium#22]
To be consistent with k8s-openapi, and general rust conventions; we should output the following:

```rust
#[serde(rename_all = "camelCase")]
pub struct ClusterSpec {
    pub cluster_network: Option<ClusterNetwork>,
    pub control_plane_endpoint: Option<ApiEndpoint>,
    pub control_plane_ref: Option<core::v1::ObjectReference>,
    pub infrastructure_ref: Option<core::v1::ObjectReference>,
    pub paused: Option<bool>,
    pub topology: Option<Topology>,
}
```

rather than what we are outputting:

```rust
pub struct ClusterSpec {
    pub clusterNetwork: Option<ClusterNetwork>,
    pub controlPlaneEndpoint: Option<ApiEndpoint>,
    pub controlPlaneRref: Option<core::v1::ObjectReference>,
    pub infrastructureRef: Option<core::v1::ObjectReference>,
    pub paused: Option<bool>,
    pub topology: Option<Topology>,
}
```
(which would require an explicit `#[allow(non_snake_case)]` override).

Doing this would require some method / library to handle the conversion from `camelCase` to `snake_case`.

**Comment by imp** [comment]
There are surprisingly many libraries for case converting

- [convert_case](https://docs.rs/convert_case)
- [heck](https://docs.rs/heck)
- [Inflector](https://docs.rs/Inflector)
- [change-case](https://docs.rs/change-case)

Anf of course serde's [own implementation](https://github.com/serde-rs/serde/blob/master/serde_derive/src/internals/case.rs), which is, I believe, not available outside of `serde`.

Also from my experience with `cluster api` sometimes blind conversion via `rename_all` won't work.
The alternative is to put `#[serde(rename = "xxx")]` on each field, but that adds a lot of noise.
Perhaps both approaches can be used together, to only override those fields that are falling out of the convention.

Another small but realistic pitfall is that serde rename and that of the other library may not agree in some corner cases and as a result invalid structure will be produced. Using `#[serde(rename = "xxx")]` on each field will avoid this, again, by adding a lot of noise.

Anyway, the produced files in most cases are not for human consumption, so maybe it is not that bad after all.

WDYT?

**Comment by clux** [comment]
> Also from my experience with cluster api sometimes blind conversion via rename_all won't work. The alternative is to put #[serde(rename = "xxx")] on each field, but that adds a lot of noise. Perhaps both approaches can be used together, to only override those fields that are falling out of the convention.

Yeah, I guess there's always the chance that we encounter something that we rename to something that serde won't rename back to what we found originally thing (e.g. `clusterIPs`-> `cluster_ips` -> `clusterIps` or something similar with multiple capitals in a row). If we can somehow detect that the path backwards won't work, and then inject a field level serde annotation as a fallback we would be good.

I don't think we could do that without inlining serde's rename implementation, but that's probably ok. It doesn't look that large, and we can document it.

**Comment by clux** [comment]
Ok, I think this just needs to be a flag:

- `--snake-case` or `-z` to toggle conversion (because it can fail on acronyms so can't default it)
- if omitted: `#[allow(non_snake_case)]` added instead
